limacharlie/net.go

```diff
@@ -0,0 +1,158 @@
+package limacharlie
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+)
+
+type NetPolicyName = string
+type NetPolicyType string
+
+var NetPolicyTypes = struct {
+	Firewall NetPolicyType
+	Service  NetPolicyType
+	Capture  NetPolicyType
+	DNS      NetPolicyType
+}{
+	Firewall: "firewall",
+	Service:  "service",
+	Capture:  "capture",
+	DNS:      "dns",
+}
+
+type NetPolicy struct {
+	CreatedBy string        `json:"created_by"`
+	ExpiresOn uint64        `json:"expires_on"`
+	Name      string        `json:"name"`
+	OID       string        `json:"oid"`
+	Type      NetPolicyType `json:"type"`
+	Policy    Dict          `json:"policy"`
+}
+
+func (n NetPolicy) jsonMarhsalContent() ([]byte, error) {
+	n.CreatedBy = ""
+	return json.Marshal(n)
+}
+
+func (n NetPolicy) EqualsContent(other NetPolicy) bool {
+	bytes, err := n.jsonMarhsalContent()
+	if err != nil {
+		return false
+	}
+	otherBytes, err := other.jsonMarhsalContent()
+	if err != nil {
+		return false
+	}
+	return string(bytes) == string(otherBytes)
+}
+
+func (n NetPolicy) WithName(name string) NetPolicy {
+	n.Name = name
+	return n
+}
+
+type NetPolicyFirewallApplicable struct {
+	TimeDayStart uint   `json:"time_of_day_start"`
+	TimeDayEnd   uint   `json:"time_of_day_end"`
+	DayWeekStart uint   `json:"day_of_week_start"`
+	DayWeekEnd   uint   `json:"day_of_week_end"`
+	Timezone     string `json:"tz"`
+}
+
+func (n NetPolicy) WithFirewallPolicy(bpfFilter string, isAllow bool, tag string, applicableTimes []NetPolicyFirewallApplicable, sources []string) NetPolicy {
+	n.Type = NetPolicyTypes.Firewall
+	n.Policy = Dict{
+		"tag":        tag,
+		"bpf_filter": bpfFilter,
+		"is_allow":   isAllow,
+		"times":      applicableTimes,
+		"sources":    sources,
+	}
+	return n
+}
+
+type NetPolicyServiceProtocolType string
+
+var NetPolicyServiceProtocolTypes = struct {
```

Comment:
Those values are integers when passed as JSON so it's probably better to keep them as `uint8`.